src/3d/nodes_container.cpp

```diff
@@ -319,9 +319,20 @@ void NodesContainer::add_or_update_text(const DebugDraw3DScopeConfig::Data *p_cf
 	opts_hash = hash_murmur3_one_float(color.a, opts_hash);
 	opts_hash = hash_murmur3_one_32(p_cfg->text_outline_color_hash, opts_hash);
 	opts_hash = hash_murmur3_one_32((uint32_t)p_cfg->text_outline_size, opts_hash);
+	opts_hash = hash_murmur3_one_32((uint32_t)p_cfg->text_fixed_size, opts_hash);
 
 	uint32_t text_hash = text.hash();
 
+	Vector2 viewport_size = p_cfg->dcd.viewport ? p_cfg->dcd.viewport->get_visible_rect().size : Vector2(1, 1);
+	Camera3D *cam = p_cfg->dcd.viewport ? p_cfg->dcd.viewport->get_camera_3d() : nullptr;
+	real_t pixel_size = 0.005;
+	if (cam) {
+		bool keep_height = cam->get_keep_aspect_mode();
+		real_t aspect_ratio = 1.0 / (keep_height ? viewport_size.y : viewport_size.x);
+		real_t fov = (cam->get_projection() == godot::Camera3D::ProjectionType::PROJECTION_PERSPECTIVE ? cam->get_fov() : 90.0);
+		pixel_size = godot::Math::atan(fov / 2) * aspect_ratio;
+	}
+
```

Comment:
```suggestion
	real_t pixel_size = 0.005f;
	if (p_cfg->text_fixed_size) {
		Vector2 viewport_size = p_cfg->dcd.viewport ? p_cfg->dcd.viewport->get_visible_rect().size : Vector2(1, 1);
		Camera3D *cam = p_cfg->dcd.viewport ? p_cfg->dcd.viewport->get_camera_3d() : nullptr;
		if (cam) {
			bool keep_height = cam->get_keep_aspect_mode();
			real_t aspect_ratio = 1.0f / (keep_height ? viewport_size.y : viewport_size.x);
			real_t fov = (cam->get_projection() == godot::Camera3D::ProjectionType::PROJECTION_PERSPECTIVE ? cam->get_fov() : 90.0);
			pixel_size = godot::Math::atan(fov * 0.5f) * aspect_ratio;
		}
	}

```

I think it's better to calculate the `pixel_size` only if necessary. And I'm not sure if `pixel_size` needs to be added to the hash. Most likely, yes.

I also added `f` for `float`s. And replaced `/ 2` with `* 0.5f`, it should be faster.